voila/handler.py

```diff
@@ -136,7 +142,10 @@ def _jinja_cell_generator(self, nb, kernel_id):
 
         with ep.setup_preprocessor(nb, resources, km=km):
             for cell_idx, cell in enumerate(nb.cells):
-                res = ep.preprocess_cell(cell, resources, cell_idx, store_history=False)
+                try:
+                    res = ep.preprocess_cell(cell, resources, cell_idx, store_history=False)
+                except:
+                    raise tornado.web.HTTPError(500, 'Execution error in cell {}'.format(cell_idx + 1))
```

Comment:
Unfortunately, this does not show the 500 page. The page is stuck on the spinner.